aspnetcore/blazor/state-management.md

```diff
@@ -0,0 +1,357 @@
+
+## Protected Browser Storage experimental package
+
+An example of a NuGet package that provides [Data Protection](xref:security/data-protection/introduction) for `localStorage` and `sessionStorage` is [Microsoft.AspNetCore.ProtectedBrowserStorage](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage).
+
+> [!WARNING]
+> `Microsoft.AspNetCore.ProtectedBrowserStorage` is an unsupported experimental package unsuitable for production use at this time.
+
+### Installation
+
+To install the `Microsoft.AspNetCore.ProtectedBrowserStorage` package:
+
+1. In the Blazor server-side app project, add a package reference to [Microsoft.AspNetCore.ProtectedBrowserStorage](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage).
+1. In the top-level HTML (for example, in the *Pages/_Host.cshtml* file in the default project template), add the following `<script>` tag:
+
+   ```html
+   <script src="_content/Microsoft.AspNetCore.ProtectedBrowserStorage/protectedBrowserStorage.js"></script>
+   ```
+
+1. In the `Startup.ConfigureServices` method, call `AddProtectedBrowserStorage` to add `localStorage` and `sessionStorage` services to the service collection:
+
+   ```csharp
+   services.AddProtectedBrowserStorage();
+   ```
+
+### Save and load data within a component
+
+In any component that requires loading or saving data to browser storage, use [@inject](xref:blazor/components#request-a-service-in-a-component) to inject an instance of either `ProtectedLocalStorage` or `ProtectedSessionStorage`, depending on which backing store you wish to use. In the following example, `sessionStorage` is used:
+
+```cshtml
+@using Microsoft.AspNetCore.ProtectedBrowserStorage
+@inject ProtectedSessionStorage ProtectedSessionStore
+```
+
+The `@using` statement can be placed into an *_Imports.razor* file instead of in the component, which makes the namespace available to larger segments of the app or the whole app.
+
+To persist the `currentCount` value in the `Counter` component of the project template, modify the `IncrementCount` method to use `ProtectedSessionStore.SetAsync`:
+
+```csharp
+private async Task IncrementCount()
+{
+    currentCount++;
+    await ProtectedSessionStore.SetAsync("count", currentCount);
+}
+```
+
+In larger, more realistic apps, storage of individual fields is an unlikely scenario. Apps are more likely to store entire model objects that include complex state. `ProtectedSessionStore` automatically JSON serializes and deserializes any data provided.
+
+In the preceding code example, the `currentCount` data is stored as `sessionStorage['count']` in the user's browser. If `sessionStorage['count']` is evaluated in the browser's developer console, the data isn't stored in plaintext but rather is protected using ASP.NET Core's [Data Protection](xref:security/data-protection/introduction).
```

Comment:
```suggestion
In the preceding code example, the `currentCount` data is stored as `sessionStorage['count']` in the user's browser. If `sessionStorage['count']` is evaluated in the browser's developer console, you'll see that the data isn't stored in plaintext but rather is protected using ASP.NET Core's [Data Protection](xref:security/data-protection/introduction).
```

... to clarify that it's not the act of evaluating this that changes how it get stored